File: code_samples/csharp/security_authorization.cs
using System;

/// <summary>
/// Security: Authorization
/// AI/ML Training Sample
/// </summary>
public class Authorization
{
    private string data;
    
    public Authorization()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new Authorization();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}

File: code_samples/go/security_authorization.go
package main

import (
    "fmt"
)

// Security: Authorization
// AI/ML Training Sample

type Authorization struct {
    Data string
}

func NewAuthorization() *Authorization {
    return &Authorization{
        Data: "",
    }
}

func (s *Authorization) Process(input string) {
    s.Data = input
}

func (s *Authorization) Validate() bool {
    return len(s.Data) > 0
}

func (s *Authorization) GetData() string {
    return s.Data
}

func main() {
    instance := NewAuthorization()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
